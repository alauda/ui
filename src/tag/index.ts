export * from './check-tag/check-tag.component';
export * from './tag.component';
export * from './tag.module';
export * from './tag.types';
