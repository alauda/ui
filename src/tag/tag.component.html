<div
  [class]="rootClass"
  [ngStyle]="customStyle"
>
  <span class="aui-tag__content"><ng-content></ng-content></span>
  <aui-icon
    *ngIf="closeable"
    class="aui-tag__close"
    icon="close_small"
    (click)="close.emit(); $event.stopPropagation()"
  ></aui-icon>
</div>
