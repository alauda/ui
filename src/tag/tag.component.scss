@use 'sass:math';
@import '../theme/var';
@import '../theme/mixin';

@mixin colorful-tag($color) {
  color: use-rgb($color);
  background-color: use-rgb($color, 6);

  &.hasBorder {
    border-color: use-rgb($color, 2);

    &.allowClick {
      &:hover {
        border-color: use-rgb($color, 1);
      }

      &:active {
        border-color: use-rgb($color, 0);
      }
    }
  }

  &.allowClick {
    &:hover {
      background-color: use-rgb($color, 7);
      cursor: pointer;
    }

    &:active {
      color: use-rgb($color);
      background-color: use-rgb($color, 6);
    }
  }
}

.aui-tag {
  display: inline-flex;
  align-items: center;
  border-radius: use-var(border-radius-m);
  border: 1px solid transparent;
  max-width: 300px;
  vertical-align: middle;
  @include transition(background-color);

  &__content {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  .aui-tag__close .aui-icon {
    margin-left: 4px;
    border-radius: 50%;
    cursor: pointer;
    @include transition;
  }

  &--large {
    padding: 0 use-var(inline-padding-m);
    height: use-var(inline-height-m);
    @include text-set(m);

    &.isRound {
      border-radius: calc(#{use-var(inline-height-m)} / 2);
    }
  }

  &--medium {
    padding: 0 use-var(inline-padding-s);
    height: use-var(inline-height-s);
    @include text-set(s);

    &.isRound {
      border-radius: calc(#{use-var(inline-height-s)} / 2);
    }
  }

  &--small {
    padding: 0 use-var(inline-padding-xs);
    height: use-var(inline-height-xs);
    @include text-set(s);

    &.isRound {
      border-radius: calc(#{use-var(inline-height-xs)} / 2);
    }
  }

  &--mini {
    padding: 0 use-var(inline-padding-xs);
    height: 20px;
    @include text-set(s);

    &.isRound {
      border-radius: calc(#{use-var(inline-height-xs)} / 2);
    }
  }

  &--primary {
    @include colorful-tag(primary);

    .aui-tag__close .aui-icon:hover {
      color: use-rgb(p-6);
      background-color: use-rgb(primary);
    }
  }

  &--success {
    @include colorful-tag(green);

    .aui-tag__close .aui-icon:hover {
      color: use-rgb(g-6);
      background-color: use-rgb(green);
    }
  }

  &--warning {
    @include colorful-tag(yellow);

    .aui-tag__close .aui-icon:hover {
      color: use-rgb(y-6);
      background-color: use-rgb(yellow);
    }
  }

  &--error {
    @include colorful-tag(red);

    .aui-tag__close .aui-icon:hover {
      color: use-rgb(r-6);
      background-color: use-rgb(red);
    }
  }

  &--info {
    color: use-rgb(n-2);
    background-color: use-rgb(n-8);

    &.hasBorder {
      border-color: use-rgb(n-7);
      background-color: use-rgb(button-bg);
    }

    .aui-tag__close .aui-icon:hover {
      color: use-rgb(n-9);
      background-color: use-rgb(n-4);
    }
  }

  &.isInvalid {
    text-decoration: line-through;
  }

  &.isSolid {
    color: white;

    .aui-tag__close .aui-icon:hover {
      color: white;
    }
  }

  &.isSolid#{&} {
    &--primary {
      background-color: use-rgb(primary);
    }

    &--success {
      background-color: use-rgb(green);
    }

    &--warning {
      background-color: use-rgb(yellow);
    }

    &--error {
      background-color: use-rgb(red);
    }

    &--info {
      background-color: use-rgb(button-bg);
    }
  }

  &.allowClick {
    a {
      color: inherit;
    }

    &:hover {
      cursor: pointer;
    }
  }
}

aui-tag + aui-tag {
  .aui-tag--large,
  .aui-tag--medium,
  .aui-tag--small {
    margin-left: use-var(spacing-m);
  }

  .aui-tag--mini {
    margin-left: 6px;
  }
}
