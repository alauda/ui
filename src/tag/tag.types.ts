import { ValueOf } from '../internal/types';

export const TagType = {
  Primary: 'primary',
  Success: 'success',
  Warning: 'warning',
  Error: 'error',
  Info: 'info',
} as const;

export type TagType = ValueOf<typeof TagType>;
