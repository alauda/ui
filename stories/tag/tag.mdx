import { Canvas, Controls, Meta } from '@storybook/blocks';

import * as basic from './basic.stories';
import * as check from './check.stories';
import * as custom from './custom.stories';
import * as icon from './icon.stories';

<Meta title="Example/Tag" />

# Tag

标签

## 基本

<Canvas
  of={basic.Basic}
  meta={basic}
></Canvas>

<Controls of={basic.Basic} />

## 可勾选

<Canvas
  of={check.Check}
  meta={check}
></Canvas>

## 自定义颜色

<Canvas
  of={custom.Custom}
  meta={custom}
></Canvas>

## Icon

<Canvas
  of={icon.Icon}
  meta={icon}
></Canvas>

## Tag Inputs

| 名称      | 类型          | 默认值               | 描述                                            |
| --------- | ------------- | -------------------- | ----------------------------------------------- |
| type      | TagType       | TagType.Primary      | 标签类型                                        |
| size      | ComponentSize | ComponentSize.Medium | 标签大小                                        |
| color     | string        | -                    | 自定义颜色, `'BORDER_COLOR[,BACKGOUNRD_COLOR]'` |
| border    | boolean       | false                | 是否添加边框                                    |
| round     | boolean       | true                 | 是否为圆角                                      |
| closeable | boolean       | false                | 是否可以关闭                                    |
| solid     | boolean       | false                | 是否为实心                                      |
| invalid   | boolean       | false                | 是否无效                                        |

## Check Tag Inputs

| 名称    | 类型          | 默认值               | 描述        |
| ------- | ------------- | -------------------- | ----------- |
| size    | ComponentSize | ComponentSize.Medium | 标签大小    |
| round   | boolean       | true                 | 是否为圆角  |
| checked | boolean       | false                | model value |

## Check Tag Outputs

| 名称          | 类型    | 默认值 | 描述               |
| ------------- | ------- | ------ | ------------------ |
| checkedChange | boolean | -      | 获取状态变化后的值 |
