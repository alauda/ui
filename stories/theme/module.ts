import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { CustomizeComponent } from './customize/component';
import { PaletteComponent } from './palette/component';

import {
  ButtonModule,
  ColorPickerModule,
  FormModule,
  IconModule,
  InputModule,
  TooltipModule,
} from '@alauda/ui';

@NgModule({
  imports: [
    CommonModule,
    ButtonModule,
    ColorPickerModule,
    IconModule,
    InputModule,
    FormModule,
    TooltipModule,
  ],
  declarations: [PaletteComponent, CustomizeComponent],
  exports: [PaletteComponent, CustomizeComponent],
})
export class ThemeModule {}
