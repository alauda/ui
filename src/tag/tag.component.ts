import {
  ChangeDetectionStrategy,
  Component,
  EventEmitter,
  Input,
  Output,
  ViewEncapsulation,
} from '@angular/core';

import { ComponentSize } from '../types';
import { Bem, buildBem } from '../utils';

import { TagType } from './tag.types';

@Component({
  selector: 'aui-tag',
  templateUrl: './tag.component.html',
  styleUrls: ['./tag.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
  encapsulation: ViewEncapsulation.None,
  preserveWhitespaces: false,
})
export class TagComponent {
  bem: Bem = buildBem('aui-tag');

  @Input()
  type: TagType = TagType.Primary;

  @Input()
  size: ComponentSize = ComponentSize.Medium;

  @Input()
  closeable = false;

  @Input()
  border = false;

  @Input()
  solid = false;

  @Input()
  invalid = false;

  @Input()
  round = true;

  @Input()
  color = '';

  @Input()
  allowClick = false;

  @Output()
  close = new EventEmitter<void>();

  get rootClass() {
    return `${this.bem.block(
      this.type || TagType.Primary,
      this.size || ComponentSize.Medium,
    )} ${this.solid ? 'isSolid' : ''} ${this.border ? 'hasBorder' : ''} ${
      this.invalid ? 'isInvalid' : ''
    } ${this.round ? 'isRound' : ''} ${
      this.allowClick && this.isNormalTag ? 'allowClick' : ''
    }`;
  }

  // 只有常规标签支持单击样式
  get isNormalTag() {
    return !this.solid && !this.closeable && !this.invalid;
  }

  get customStyle() {
    const arr = this.color.split(',');
    return {
      color: arr[0],
      'border-color': arr[0],
      'background-color': arr[1] || arr[0],
    };
  }
}
