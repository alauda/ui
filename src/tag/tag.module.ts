import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { IconModule } from '../icon';

import { CheckTagComponent } from './check-tag/check-tag.component';
import { TagComponent } from './tag.component';

@NgModule({
  imports: [CommonModule, IconModule],
  declarations: [TagComponent, CheckTagComponent],
  exports: [TagComponent, CheckTagComponent],
})
export class TagModule {}
