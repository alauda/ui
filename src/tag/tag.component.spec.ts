import { Component } from '@angular/core';
import { TestBed } from '@angular/core/testing';

import { TagModule } from './tag.module';

describe('TagComponent', () => {
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [TagModule],
      declarations: [TestComponent],
    });
  });

  it('should match snapshot', () => {
    const fixture = TestBed.createComponent(TestComponent);
    expect(fixture).toMatchSnapshot();
  });
});

@Component({
  template: `
    <aui-tag>primary</aui-tag>
    <aui-tag
      type="success"
      [size]="'medium'"
      [solid]="true"
      [invalid]="true"
    >
      success
    </aui-tag>
    <aui-tag
      type="warning"
      [size]="'small'"
      [closeable]="true"
    >
      warning
    </aui-tag>
    <aui-tag
      type="error"
      [size]="'mini'"
      [round]="true"
      >error</aui-tag
    >
    <aui-tag
      type="info"
      size="mini"
      [solid]="true"
      [closeable]="true"
      [invalid]="true"
      [round]="true"
    >
      info
    </aui-tag>
  `,
})
class TestComponent {}
