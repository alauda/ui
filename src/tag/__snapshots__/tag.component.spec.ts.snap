// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagComponent should match snapshot 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
>
  <aui-tag>
    <div>
      <span
        class="aui-tag__content"
      >
        primary
      </span>
    </div>
  </aui-tag><aui-tag
    type="success"
  >
    <div>
      <span
        class="aui-tag__content"
      >
         success 
      </span>
    </div>
  </aui-tag><aui-tag
    type="warning"
  >
    <div>
      <span
        class="aui-tag__content"
      >
         warning 
      </span>
    </div>
  </aui-tag><aui-tag
    type="error"
  >
    <div>
      <span
        class="aui-tag__content"
      >
        error
      </span>
    </div>
  </aui-tag><aui-tag
    size="mini"
    type="info"
  >
    <div>
      <span
        class="aui-tag__content"
      >
         info 
      </span>
    </div>
  </aui-tag>
</ng-component>
`;
